contracts/strategies/StrategyTUSDypool.sol

```diff
@@ -0,0 +1,365 @@
+            ICurveFi(ypool).exchange(1, 3, _yusdc, tmp.mul(DENOMINATOR.sub(slip)).div(DENOMINATOR));
+        }
+        if (_yusdt > 0) {
+            IERC20(yusdt).safeApprove(ypool, 0);
+            IERC20(yusdt).safeApprove(ypool, _yusdt);
+            tmp = ICurveFi(ypool).get_dy(2, 3, _yusdt);
+            ICurveFi(ypool).exchange(2, 3, _yusdt, tmp.mul(DENOMINATOR.sub(slip)).div(DENOMINATOR));
+        }
+
+        uint256 _before = IERC20(want).balanceOf(address(this));
+        yvERC20(ytusd).withdraw(IERC20(ytusd).balanceOf(address(this)));
+        uint256 _after = IERC20(want).balanceOf(address(this));
+
+        return _after.sub(_before);
+    }
+
+    // Withdraw all funds, normally used when migrating strategies
+    function withdrawAll() external returns (uint256 balance) {
+        require(msg.sender == controller, "!controller");
+        _withdrawAll();
+
+        balance = IERC20(want).balanceOf(address(this));
+
+        address _vault = IController(controller).vaults(address(want));
+        require(_vault != address(0), "!vault"); // additional protection so we don't burn the funds
+        IERC20(want).safeTransfer(_vault, balance);
+    }
+
+    function _withdrawAll() internal {
+        uint256 _yycrv = IERC20(yycrv).balanceOf(address(this));
+        if (_yycrv > 0) {
+            yvERC20(yycrv).withdraw(_yycrv);
+            _withdrawOne(IERC20(ycrv).balanceOf(address(this)));
+        }
+    }
+
+    function balanceOfWant() public view returns (uint256) {
+        return IERC20(want).balanceOf(address(this));
+    }
+
+    function balanceOfYCRV() public view returns (uint256) {
+        return IERC20(ycrv).balanceOf(address(this));
+    }
+
+    function balanceOfYCRVinWant() public view returns (uint256) {
+        return balanceOfYCRV().mul(ICurveFi(ypool).get_virtual_price()).div(1e18);
+    }
+
+    function balanceOfYYCRV() public view returns (uint256) {
+        return IERC20(yycrv).balanceOf(address(this));
+    }
+
+    function balanceOfYYCRVinYCRV() public view returns (uint256) {
+        return balanceOfYYCRV().mul(yvERC20(yycrv).getPricePerFullShare()).div(1e18);
+    }
+
+    function balanceOfYYCRVinWant() public view returns (uint256) {
+        return balanceOfYYCRVinYCRV().mul(ICurveFi(ypool).get_virtual_price()).div(1e18);
+    }
+
+    function balanceOf() public view returns (uint256) {
+        return balanceOfWant().add(balanceOfYYCRVinWant());
+    }
+
+    function migrate(address _strategy) external {
+        require(msg.sender == governance, "!governance");
+        require(IController(controller).approvedStrategies(want, _strategy), "!stategyAllowed");
+        IERC20(yycrv).safeTransfer(_strategy, IERC20(yycrv).balanceOf(address(this)));
+        IERC20(ycrv).safeTransfer(_strategy, IERC20(ycrv).balanceOf(address(this)));
+        IERC20(want).safeTransfer(_strategy, IERC20(want).balanceOf(address(this)));
+    }
+
+    function forceD(uint256 _amount) external isAuthorized {
```

Comment:
is this method for an emergency to force a deposit? in what case would this be called?

Reply:
To make it less bearish. In a special case when curve pool is suddenly skyrocketing after a dynamic cap is touched. IN normal case, this won't be triggered. Or I could write a keeper bot to auto-monitor this.